### ISSUE TITLE
Fatal error is thrown w/ Elementor Pro

### ISSUE THREAD

**Issue by tomalec** [woocommerce/woocommerce-google-analytics-integration#266]
### Describe the bug:

Merchant reported a problem when GA 1.7.1 is run with Elementor Pro (6219949-zd-woothemes)

```
2023-04-26T10:04:38+00:00 CRITICAL Klaidos žinutė: Uncaught Error: Call to a member function is_type() on bool in /home/_site_/public_html/wp-content/plugins/woocommerce-google-analytics-integration/includes/class-wc-google-gtag-js.php:52
Stack trace:
#0 /home/_site_/public_html/wp-includes/class-wp-hook.php(308): WC_Google_Gtag_JS->setup_frontend_scripts()
#1 /home/_site_/public_html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters()
#2 /home/_site_/public_html/wp-includes/plugin.php(517): WP_Hook->do_action()
#3 /home/_site_/public_html/wp-content/plugins/elementor-pro/modules/woocommerce/documents/product.php(117): do_action()
#4 /home/_site_/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(156): ElementorPro\Modules\Woocommerce\Documents\Product->before_get_content()
#5 /home/_site_/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/documents/theme-document.php(173): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document->get_content()
#6 /home/_site_/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/documents/single-base.php(93): ElementorPro\Modules\ThemeBuilder\Documents\Theme_Document->print_content()
#7 /home/_site_/public_html/wp-content/plugins/elementor-pro/modules/woocommerce/documents/product.php(133): ElementorPro\Modules\ThemeBuilder\Documents\Single_Base->print_content()
#8 /home/_site_/public_html/wp-content/plugins/elementor-pro/modules/theme-builder/classes/locations-manager.php(315): ElementorPro\Modules\Woocommerce\Documents\Product->print_content()
#9 /home/_site_/public_html/wp-content/themes/astra/inc/compatibility/class-astra-elementor-pro.php(220): ElementorPro\Modules\ThemeBuilder\Classes\Locations_Manager->do_location()
#10 /home/_site_/public_html/wp-includes/class-wp-hook.php(308): ElementorPro\Modules\ThemeBuilder\ThemeSupport\Astra_Elementor_Pro->do_template_parts()
#11 /home/_site_/public_html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters()
#12 /home/_site_/public_html/wp-includes/plugin.php(517): WP_Hook->do_action()
#13 /home/_site_/public_html/wp-content/themes/astra/inc/class-astra-loop.php(191): do_action()
#14 /home/_site_/public_html/wp-includes/class-wp-hook.php(308): Astra_Loop->loop_markup()
#15 /home/_site_/public_html/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters()
#16 /home/_site_/public_html/wp-includes/plugin.php(517): WP_Hook->do_action()
#17 /home/_site_/public_html/wp-content/themes/astra/inc/core/theme-hooks.php(258): do_action()
#18 /home/_site_/public_html/wp-content/themes/astra/single.php(27): astra_content_loop()
#19 /home/_site_/public_html/wp-includes/template-loader.php(106): include('/home/u56680679...')
#20 /home/_site_/public_html/wp-blog-header.php(19): require_once('/home/u56680679...')
#21 /home/_site_/public_html/index.php(17): require('/home/u56680679...')
#22 {main}
  thrown (rasta faile /home/_site_/public_html/wp-content/plugins/woocommerce-google-analytics-integration/includes/class-wc-google-gtag-js.php, eilutės numeris 52)
  ```

### Steps to reproduce:

1.  > I get a fatal error when I open WP – Templates and click view. Otherwise, this error does not occur. Disabling your plugin also clears this error.

### Expected behavior:

No errors are thrown

### Actual behavior:

Fatal is thrown

### Additional details:

They suggest
Changing

> ```php
> if ( $product->is_type( 'variable' ) ) { `
> ```
> To
> ```php
> if ( is_a( $product, 'WC_Product_Variable' ) ) {
> ```

Indeed when setting up frontend scripts in https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/trunk/includes/class-wc-google-gtag-js.php#L49-L52
we make no assertion on the `global $product` variable type.

So changing that to `is_a` should prevent as from throwing a fatal error. 
The only doubts I have are:
1. What condition under which `$product` is not what we expected, why, and what other issue may we hide by such a fix?
3. Should we also assert the `global $product` type in all the other places?

The code was introduced in https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/230/files#diff-75fada6b677c39f17a7de173a315db4d8911e7469f5a70dce4822a6aaa849514R45-R47
//cc @martynmjones

**Comment by mikkamp** [comment]
In the initial report it was mentioned:

> I get a fatal error when I open WP – Templates and click view. Otherwise, this error does not occur. Disabling your plugin also clears this error.

Since this is a conflict with Elementor Pro. It sounds like opening the templates (which is a backend request) is triggering the generation of the page (maybe a preview), but not providing the full context (such as setting the product variable).
While we can get around it by checking if a product exists, shouldn't we also ensure that in cases like this the tracking code doesn't get injected on the page at all? We don't want to be tracking editing the template as a page view.

**Comment by sukafia** [comment]
@mikkamp you're absolutely correct, opening an Elementor template in the WP Admin triggers the error.  Here's a video of the steps to replicate it: https://youtube.com/shorts/qel4_W5CIPo?feature=share

**Comment by tomalec** [comment]
The fix should be released next week as `1.8.1`. We didn't manage to reproduce locally fully, so we are not sure about any other unexpected behavior. But the reported fatal error should be solved.

Please reopen or report a new issue if there are any further problems.
